pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2063,28 +2070,76 @@ public boolean addNewReloadAllSegmentsJob(String tableNameWithType, String jobId
         ZKMetadataProvider.constructPropertyStorePathForControllerJob(ControllerJobType.RELOAD_SEGMENT));
   }
 
+  /**
+   * Adds a new job metadata for controller job like table rebalance or reload into ZK
+   * @param jobId job's UUID
+   * @param jobMetadata the job metadata
+   * @param jobResourcePath zkPath to add the new job metadata
+   * @return boolean representing success / failure of the ZK write step
+   */
   public boolean addControllerJobToZK(String jobId, Map<String, String> jobMetadata, String jobResourcePath) {
```

Comment:
Not introduced in this PR, but suggest changing the last argument to `ControllerJobType`

Reply:
sgtm. will refactor this here.